src/sanitize-argument.ts

```diff
@@ -107,6 +122,30 @@ export function sanitize_redis(argv:any)
   }
 }
 
+export async function checkApiAvailabilty(url: string): Promise<boolean>{
+  try {
+    const resp = await axios.get(url, { maxRedirects: 0 });
+    return resp.status === 200 && !resp.headers['mediawiki-api-error'];
+  } catch (err) {
+    return false;
+  }
+}
```

Comment:
Full code duplicate, this should never happen.

Reply:
I don't think you need that anyway, you just need to make a simple get request on a URL and check if code is HTTP 200. You can do that directly with Axios.